framework/wazuh/agent.py

```diff
@@ -325,42 +344,28 @@ def get_agents(agent_list: list = None, offset: int = 0, limit: int = common.DAT
         Select fields to return. Format: {"fields":["field1","field2"]}.
     sort : dict
         Sorts the items. Format: {"fields":["field1","field2"],"order":"asc|desc"}.
-    search : dict
-        Look for elements with the specified string. Format: {"fields": ["field1","field2"]}
-    filters : dict
-        Defines required field filters. Format: {"field1":"value1", "field2":["value2","value3"]}
-    q : str
-        Query to filter results by.
-    distinct : bool
-        Look for distinct values.
 
     Returns
     -------
     AffectedItemsWazuhResult
         Affected items.
     """
-    result = AffectedItemsWazuhResult(all_msg='All selected agents information was returned',
-                                      some_msg='Some agents information was not returned',
-                                      none_msg='No agent information was returned'
-                                      )
-    if agent_list:
-        if filters is None:
-            filters = dict()
-
-        system_agents = get_agents_info()
-
-        for agent_id in agent_list:
-            if agent_id not in system_agents:
-                result.add_failed_item(id_=agent_id, error=WazuhResourceNotFound(1701))
+    result = AffectedItemsWazuhResult(
+        all_msg='All selected agents information was returned',
+        some_msg='Some agents information was not returned',
+        none_msg='No agent information was returned'
+    )
+    if filters is None:
+        filters = dict()
 
-        rbac_filters = get_rbac_filters(system_resources=system_agents, permitted_resources=agent_list, filters=filters)
+    query = build_agents_query(agents_id, filters)
 
-        with WazuhDBQueryAgents(offset=offset, limit=limit, sort=sort, search=search, select=select,
-                                query=q, **rbac_filters, distinct=distinct) as db_query:
-            data = db_query.run()
+    async with get_indexer_client() as indexer:
+        items = await indexer.agents.search(query, select=select, exclude='key', limit=limit, offset=offset, sort=sort)
+        items = [i for i in get_source_items(items)]
```

Comment:
I did the change for retrieve the groups as a list.

Regarding the ID, I'll add a note in https://github.com/wazuh/wazuh/issues/25582 to check it.

Reply:
I asked in the protocol channel to see how we proceed since we did different things in two PRs that are pending review and related to the same code.

Based on the response we could decide to get it from the source or change it to get it from the document ID